roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,137 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class DockerImageAvailability(OpenShiftCheck):
+    """Check that required Docker images are available.
+
+    This check attempts to pull required docker images and fail if
+    any image is missing or an error occurs during pull.
+    """
+
+    name = "docker_image_availability"
+    tags = ["preflight"]
+
+    skopeo_image = "juanvallejo/skopeo:latest"
+
+    registry_prefix = {
+        "origin": "openshift/origin",
+        "openshift-enterprise": "openshift3/ose",
+    }
+
+    def run(self, _, task_vars):
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        openshift_release = get_var(task_vars, "openshift_release")
+        openshift_image_tag = get_var(task_vars, "openshift_image_tag")
+        deployment_type = get_var(task_vars, "deployment_type")
+
+        reg_prefix = self.registry_prefix[deployment_type]
+        required_images = self.get_required_images(is_containerized, reg_prefix, openshift_release)
+
+        # append images with qualified image tags to our list of required images.
+        # these are images with a (v0.0.0.0) tag, rather than a standard release
+        # format tag (v0.0). We want to check this set in both containerized and
+        # non-containerized installations.
+        required_images.update(self.qualified_docker_images(reg_prefix, "v" + openshift_image_tag))
+
+        # use docker_image_facts to inspect local images before attempting to use Skopeo
+        required_images = self.inspect_local_images(required_images, task_vars)
+
+        # ensure skopeo image exists or can be pulled and used successfully
+        failed = self.check_skopeo_image(task_vars)
+        if failed:
+            msg = "Failed to pull or use existing Skopeo image: {msg}".format(msg=failed)
+            return {"failed": True, "msg": msg}
+
+        # use docker info to determine which registries
+        # Skopeo should inspect for required images
+        docker_info = self.get_docker_info(task_vars)
+        if not docker_info:
+            return {"failed": True, "msg": "Unable to get docker host info. Is `docker-py` installed?"}
+
+        docker_regs = docker_info.get("Registries")
+        if not docker_regs:
+            return {"failed": True, "msg": "Unable to read docker registry data from host."}
+
+        regs = [reg["Name"] for reg in docker_regs]
+
+        # inspect all rpm or containerized docker images for every docker registry found
+        while regs and required_images:
+            current_reg = regs.pop()
```

Comment:
Yeah, this is much better than the previous :-)

Naming suggestion:

s/docker_regs/registries
s/reg/registry
s/current_reg/registry

Note we don't use `docker_regs` apart from checking for emptiness, so we can write:

```python
registries = [registry.get("Name", "") for registry in docker_info.get("Registries", {})]

if not registries:
    return ... # failed

while registries and required_images:
    registry = registries.pop(0)
    required_images = ...
```

Note: I suspect we should pop items from the front of the list (`pop(0)` x `pop()`) so as to try each registry in the proper order.